torch/csrc/utils/python_arg_parser.h

```diff
@@ -291,6 +297,9 @@ inline double PythonArgs::toDoubleWithDefault(int i, double default_double) {
 }
 
 inline bool PythonArgs::toBool(int i) {
+  if (!signature.params[i].optional) {
+    return args[i] == Py_True;
+  }
   return toBoolWithDefault(i, signature.params[i].default_bool);
```

Comment:
Please add a unit test somewhere!

Reply:
Do we have tests for the python arg parser? I couldn't find any.

Reply:
Sorry, I don't really understand what the bug was. Is it from the following:
- users were not providing optional arguments
- the native_functions.yaml declarations were missing default arguments
?

Reply:
I agree, the linked issue is not clear.